lite/proxy/validate_test.go

```diff
@@ -0,0 +1,256 @@
+package proxy_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/tendermint/tendermint/lite"
+	"github.com/tendermint/tendermint/lite/proxy"
+	"github.com/tendermint/tendermint/types"
+)
+
+var (
+	deabBeefTxs           = types.Txs{[]byte("DE"), []byte("AD"), []byte("BE"), []byte("EF")}
+	deadBeefRipEmd160Hash = deabBeefTxs.Hash()
+)
+
+func TestValidateBlock(t *testing.T) {
+	tests := []struct {
+		block   *types.Block
+		commit  lite.Commit
+		wantErr string
+	}{
+		{
+			block: nil, wantErr: "non-nil Block",
+		},
+		{
+			block: &types.Block{}, wantErr: "nil Header",
+		},
+		{
+			block: &types.Block{Header: new(types.Header)},
+		},
+
+		// Start Header.Height mismatch test
+		{
+			block:   &types.Block{Header: &types.Header{Height: 10}},
+			commit:  lite.Commit{Header: &types.Header{Height: 11}},
+			wantErr: "don't match - 10 vs 11",
+		},
+
+		{
+			block:  &types.Block{Header: &types.Header{Height: 11}},
+			commit: lite.Commit{Header: &types.Header{Height: 11}},
+		},
+		// End Header.Height mismatch test
+
+		// Start Header.Hash mismatch test
+		{
+			block: &types.Block{
+				Header: &types.Header{
+					Height:         11,
+					Time:           time.Date(2018, 1, 1, 1, 1, 1, 1, time.UTC),
+					ValidatorsHash: []byte("Tendermint"),
+				},
+			},
+			commit:  lite.Commit{Header: &types.Header{Height: 11}},
+			wantErr: "Headers don't match",
+		},
+
+		{
+			block: &types.Block{
+				Header: &types.Header{
+					Height:         11,
+					Time:           time.Date(2018, 1, 1, 1, 1, 1, 1, time.UTC),
+					ValidatorsHash: []byte("Tendermint"),
+				},
+			},
+			commit: lite.Commit{
+				Header: &types.Header{
+					Height:         11,
+					Time:           time.Date(2018, 1, 1, 1, 1, 1, 1, time.UTC),
+					ValidatorsHash: []byte("Tendermint"),
+				},
+			},
+		},
+		// End Header.Hash mismatch test
+
+		// Start Header.Data hash mismatch test
+		{
+			block: &types.Block{
+				Header: &types.Header{Height: 11},
+				Data:   &types.Data{Txs: []types.Tx{[]byte("0xDE"), []byte("AD")}},
+			},
+			commit: lite.Commit{
+				Header: &types.Header{Height: 11},
+				Commit: &types.Commit{BlockID: types.BlockID{Hash: []byte("0xDEADBEEF")}},
+			},
+			wantErr: "Data hash doesn't match header",
+		},
+		{
+			block: &types.Block{
+				Header: &types.Header{Height: 11, DataHash: deadBeefRipEmd160Hash},
+				Data:   &types.Data{Txs: deabBeefTxs},
+			},
+			commit: lite.Commit{
+				Header: &types.Header{Height: 11},
+				Commit: &types.Commit{BlockID: types.BlockID{Hash: []byte("DEADBEEF")}},
+			},
+		},
+		// End Header.Data hash mismatch test
+	}
+
+	assert := assert.New(t)
```

Comment:
we decided not to use this contruction since it messes with linters

Reply:
Oh I see, thank you!